### ISSUE TITLE
MMClassification Plan in 2022

### ISSUE THREAD

**Issue by mzr1996** [open-mmlab/mmpretrain#678]
To develop a better image classification toolbox for everyone. Welcome any contributions and suggestions.

In this issue, **you can either**:

- Suggest a new feature by leaving a comment. The plan list will be updated in time.
- Vote for a feature request with +1 or be against with -1. More votes mean a quicker response.
- Tell us that you would like to help implement one of the features in the list or review the PRs. (This is the greatest thing to hear about!)

The following is the planned feature in 2022:

## Update Model Zoo

We plan to support more than 40 classic/new algorithms in 2022 and provide more pre-trained models.
Here is the model list, and it will be updated at any time. Propose by leaving a comment :wink: 
- [x] EfficientNet #649 
- [x] ConvNeXt #670
- [x] Twins PVPVT/SVT #pcpvt #642
- [x] HRNet #660
- [x] ResNet V1C #692
- [x] RepMLP #709
- [x] DenseNet
- [x] CSPNet #735 
- [x] Wide ResNet #715 
- [x] BeiT
- [ ] RMNet #549
- [x] ConvMixer #716
- [x] VAN 
- [x] PoolFormer
- [ ] GhostNet
- [x] inception_v3
- [x] mViT
- [x] MobileOne
...

## Support downstream tasks

How to use MMCls's backbones in MMDet or MMSeg? Can we use modules across different OpenMMLab repositories smoothly?
Yes, we do support using modules from other repositories directly in config files. But there are still some restrictions, like model variants, dynamic input images shape, and tutorials.

- [ ] Tutorial about how to apply one backbone network on classification, detection, and segmentation.
- [x] Support dynamic input images shape on transformer-based networks #706 
...

## Support more datasets and transforms
ImageNet and CIFAR are not enough! MMClassification can be used on more datasets but reinventing the wheel is troubling. What other dataset do you think mmcls can support officially? Comment or contribute your implementation directly!

- [x] CUB dataset #703 
- [x] Stanford Cars
- [x] Generic custom dataset
- [ ]  TrivialAugmentation
...

## Provide more tools to help development

Not only a model hub but also a toolbox. We have provided a series of analysis and visualization tools, like pipeline visualization and CAM visualization. What other tools do you need? Just comment!

- [x] K-fold cross-validation tools. #560 
- [x] Test-Time Augmentation #716
- [x] Multi-task training (One backbone with multi heads) #675
...

## Keep in contact with contributors

We're exploring how to communicate with contributors. Mail-list, Skype, Slack, or GitHub projects? Anyone who wants to join the contributor groups is welcome to leave a comment to tell me about your favorite contact method.

**Comment by Bo396543018** [comment]
Hope to add TrivialAugmentation，https://github.com/automl/trivialaugment

**Comment by ioir123ju** [comment]
Wave-MLP,https://github.com/huawei-noah/CV-Backbones.git

**Comment by okideal** [comment]
hope to add reid-related structures

**Comment by mzr1996** [comment]
> hope to add reid-related structures

What structures? ReID sounds should be in MMTrack

**Comment by wang11wang** [comment]
> > hope to add reid-related structures
> 
> What structures? ReID sounds should be in MMTrack

Maybe Metric Learning like triplet loss, and associated with Dataset (output a tuple image and one label)

**Comment by ioir123ju** [comment]
Is there a plan to add wave-MLP.It has amazing performance

**Comment by bmxbmx3** [comment]
another question,if I want to train on distributed model,should I customize a sampler that inherits from `DistributedSampler` class or just from `torch.utils.data.sampler.Sampler` class?

**Comment by gachiemchiep** [comment]
Is there any plan for ArcFace loss?

**Comment by Ezra-Yu** [comment]
> Hope to add TrivialAugmentation，https://github.com/automl/trivialaugment

Yes， this is in our latest plan.

**Comment by Ezra-Yu** [comment]
> Is there any plan for ArcFace loss?

Yes, We will add it in the near future.

**Comment by FabianSchuetze** [comment]
Thanks for this wonderful repo. Is there any desire to add MaxViT? I think it would be a wonderful backbone other mmlab repos could benefit from, in particular mmdetection.
